homeassistant/components/mqtt/debug_info.py

```diff
@@ -157,6 +157,80 @@ def remove_trigger_discovery_data(hass, discovery_hash):
     hass.data[DATA_MQTT_DEBUG_INFO]["triggers"][discovery_hash]["discovery_data"] = None
 
 
+def _info_for_entity(hass: HomeAssistant, entity_id: str) -> dict[str, Any]:
+    mqtt_debug_info = hass.data[DATA_MQTT_DEBUG_INFO]
+    entity_info = mqtt_debug_info["entities"][entity_id]
+    subscriptions = [
+        {
+            "topic": topic,
+            "messages": [
+                {
+                    "payload": str(msg.payload),
+                    "qos": msg.qos,
+                    "retain": msg.retain,
+                    "time": msg.timestamp,
+                    "topic": msg.topic,
+                }
+                for msg in list(subscription["messages"])
+            ],
+        }
+        for topic, subscription in entity_info["subscriptions"].items()
+    ]
+    transmitted = [
+        {
+            "topic": topic,
+            "messages": [
+                {
+                    "payload": str(msg.payload),
+                    "qos": msg.qos,
+                    "retain": msg.retain,
+                    "time": msg.timestamp,
+                    "topic": msg.topic,
+                }
+                for msg in list(subscription["messages"])
+            ],
+        }
+        for topic, subscription in entity_info["transmitted"].items()
+    ]
+    discovery_data = {
+        "topic": entity_info["discovery_data"].get(ATTR_DISCOVERY_TOPIC, ""),
+        "payload": entity_info["discovery_data"].get(ATTR_DISCOVERY_PAYLOAD, ""),
+    }
+
+    return {
+        "entity_id": entity_id,
+        "subscriptions": subscriptions,
+        "discovery_data": discovery_data,
+        "transmitted": transmitted,
+    }
+
+
+def _info_for_trigger(hass: HomeAssistant, trigger_key: str) -> dict[str, Any]:
+    mqtt_debug_info = hass.data[DATA_MQTT_DEBUG_INFO]
+    trigger = mqtt_debug_info["triggers"][trigger_key]
+    discovery_data = None
+    if trigger["discovery_data"] is not None:
+        discovery_data = {
+            "topic": trigger["discovery_data"][ATTR_DISCOVERY_TOPIC],
+            "payload": trigger["discovery_data"][ATTR_DISCOVERY_PAYLOAD],
+        }
+    return {"discovery_data": discovery_data, "trigger_key": trigger_key}
+
+
+def info_for_config_entry(hass):
+    """Get debug info for all entities and triggers."""
+    mqtt_info = {"entities": [], "triggers": []}
```

Comment:
Would it be possible to include the config entry source? That way we know if it's configured via YAML or using the add-on. 

And maybe also include if it's using SSL and the port?